app/controllers/devise_token_auth/registrations_controller.rb

```diff
@@ -6,16 +6,8 @@ class RegistrationsController < DeviseTokenAuth::ApplicationController
     skip_after_action :update_auth_header, only: [:create, :destroy]
 
     def create
-      @resource            = resource_class.new(sign_up_params.except(:confirm_success_url))
-      @resource.provider   = provider
-
-      # honor devise configuration for case_insensitive_keys
-      if resource_class.case_insensitive_keys.include?(:email)
-        @resource.email = sign_up_params[:email].try :downcase
-      else
-        @resource.email = sign_up_params[:email]
-      end
-
+      build_resource
```

Comment:
Consider removing this explicit call to `build_resource` from the `#create` action, and instead call it with `before_action :build_resource, only: [:create]`.

Reply:
@Evan-M we discussed that with @MaicolBen and i had it the 'callback way' before, but then we moved it to an explicit call, which kind of makes sense when looking at [devise](https://github.com/plataformatec/devise/blob/master/app/controllers/devise/registrations_controller.rb#L17)

Reply:
@Marinlemaignan Oops, I _should_ have been more diligent in reading through the discussion in #1143. Carry on! 🎒